lib/core/src/Cardano/Wallet/Network.hs

```diff
@@ -90,18 +86,18 @@ waitForConnection nw policy = do
         Right _ -> return ()
         Left e -> throwIO e
   where
-
     shouldRetry _ = \case
-        Right _ -> do
+        Right _ ->
             return False
-        Left (ErrNetworkTipNetworkUnreachable _) -> do
-             TIO.putStrLn "[INFO] waiting for connection to the node..."
-             return True
-        Left _ -> return True
+        Left ErrNetworkTipNotFound ->
+            return True
+        Left (ErrNetworkTipNetworkUnreachable _) ->
+            return True
```

Comment:
You said this was a slight change in behaviour — but it isn't (aside from removing the logging)?

Reply:
It now retries until the tip is found. It used to only wait for the connection to be established.

Reply:
Previously the there was a `Left _ -> True`:
```haskell
        Right _ -> do
            return False
        Left (ErrNetworkTipNetworkUnreachable _) -> do
             TIO.putStrLn "[INFO] waiting for connection to the node..."
             return True
        Left _ -> return True
```
which I admit was very weird as it was only logging in the `ErrNetworkTipNetworkUnreachable _` case.

But regardless 👍

Reply:
Hmm.. Weird. I didn't see this case on my local branch. Either I wasn't up-to-date when making the change or, I can't see things right ^^"
Anyway, no major behavioral change then.